# Analisis Financiero

En el presente notebook se cargaran los datos de el balance Sheet de Grupo Aeroportuario del Pacifico SAB de CV 'GAP'

Authors:
- Sofia Hernandez
- Alejandra Figueroa
- Luis Soto
- Luis Robles
- Manuel Reyna

### Importamos librerias y cargamos datos

In [2]:
import pandas as pd

In [58]:
# Cargar el archivo CSV
data = pd.read_csv("balance gap.csv", skiprows=7) # Tomamos a partir de la fila 8 pues ahi vienen los años

data = data.iloc[1:] #eliminamos la primera fila pues estaba llena de NaNs

#haremos el analisis a partir de 2021, por lo que eliminamos los demas años
data = data.iloc[:, :-6]

# Renombrar la primera columna
data.rename(columns={data.columns[0]: "Cuentas"}, inplace=True)

# Convertir las filas de valores (excepto la columna 'Cuentas') a numérico
data.iloc[:, 1:] = data.iloc[:, 1:].apply(pd.to_numeric, errors='coerce')

data

Cuentas DEC '24 DEC '23 DEC '22  \
1                                              Assets     NaN     NaN     NaN   
2                       Cash & Short-Term Investments   647.6   593.8   634.8   
3                                           Cash Only   647.6   593.8   634.8   
4                              Short-Term Receivables   191.8   205.7   153.7   
5                           Accounts Receivables, Net   129.7   132.9   121.5   
..                                                ...     ...     ...     ...   
69           Total Liabilities & Shareholders' Equity     NaN     NaN     NaN   
70                                          Per Share     NaN     NaN     NaN   
71                               Book Value per Share     NaN   23.12   18.93   
72                      Tangible Book Value per Share     NaN  -26.75  -14.88   
73  All figures in millions of U.S. Dollar except ...     NaN     NaN     NaN   

   DEC '21  
1      NaN  
2    651.5  
3    651.5  
4    137.4  
5     84.1  
..     ...  
69     NaN  
70     NaN  
71    18.4  
72  -10.27  
73     NaN  

[73 rows x 5 columns]

## Razones de Liquidez

Las razones de liquidez son indicadores financieros que miden la capacidad de una empresa para cumplir con sus obligaciones de corto plazo utilizando sus activos líquidos. 

Estas razones ayudan a evaluar la solvencia de una empresa y su capacidad para pagar deudas a corto plazo sin necesidad de vender activos a largo plazo o buscar financiamiento adicional.  

1. **Razón Circulante**:  
   Mide la capacidad de la empresa para cubrir sus pasivos circulantes con sus activos corrientes.  
   $$
   \text{Razón Circulante} = \frac{\text{Activo Circulante}}{\text{Pasivo Circulante}}
   $$  

2. **Prueba Ácida (Quick Ratio)**:  
   Excluye los inventarios de los activos corrientes para medir la liquidez más estricta.  
   $$
   \text{Prueba Ácida} = \frac{\text{Activo Circulante} - \text{Inventarios}}{\text{Pasivo Curculante}}
   $$  

3. **Razón de Efectivo**:
    Sirve para ver que tan liquidas son las inversiones a corto plazo.
   $$
\text{Razón de Efectivo} = \frac{\text{Dinero en Caja}}{\text{Pasivo Curculante}}
   $$


5. **Razón de Capital de Trabajo**:  
   Evalúa la diferencia entre activos y pasivos corrientes.  
   $$
   \text{Capital de Trabajo} = \text{Activo Corriente} - \text{Pasivo Corriente}
   $$  

#### Razón circulante

In [79]:
assets = data[data['Cuentas'].str.contains('Total Current Assets', case=False, na=False)]
liabilities = data[data['Cuentas'].str.contains('Total Current Liabilities', case=False, na=False)]

# tomar los valores que necesitamos para las razones
total_assets = assets.iloc[0, 1:]
total_liabilities = liabilities.iloc[0, 1:]

# calcular la razon
razon_circ = total_assets / total_liabilities

# resultados
print("Razón Criculante para cada año:")
for year, ratio in zip(razon_circ.index, razon_circ):
    print(f"{year}: {ratio}")

Razón Criculante para cada año:
DEC '24: 0.8548157198126655
DEC '23: 1.1343701835505113
DEC '22: 2.241622078287806
DEC '21: 1.751256830601093


En 2024, la razón baja a 0.85, lo que significa que la empresa tiene menos activos para cubrir sus deudas a corto plazo.
En 2023, 2022 y 2021, la razón fue mayor a 1, lo que indica que la empresa estaba en mejor situación para pagar sus deudas inmediatas.
La caída en 2024 sugiere una posible preocupación por la liquidez.

#### Prueba Acida

In [85]:
inventories = data[data['Cuentas'].str.contains('Inventories', case=False, na=False)]
inventories

Empty DataFrame
Columns: [Cuentas, DEC '24, DEC '23, DEC '22, DEC '21]
Index: []

Como podemos GAP (Grupo Aeroportuario del Pacífico) no tiene inventarios porque es una empresa de servicios dedicada a la operación de aeropuertos, no a la venta de productos físicos.

#### Razon de Efectivo

In [95]:
cash = data[data['Cuentas'].str.contains('Cash Only', case=False, na=False)]
cash

Cuentas DEC '24 DEC '23 DEC '22 DEC '21
3  Cash Only   647.6   593.8   634.8   651.5

Para esta razon vamos a tomar el Cash ya que no sabemos que tan liquidaz son sus inversiones a corto plazo.

In [100]:
total_cash = cash.iloc[0, 1:]

razon_efectivo = total_cash / total_liabilities

print("Razón de Efectivo para cada año:")
for year, ratio in zip(razon_efectivo.index, razon_efectivo):
    print(f"{year}: {ratio}")

Razón de Efectivo para cada año:
DEC '24: 0.6593361840765628
DEC '23: 0.8320022418383073
DEC '22: 1.7876654463531396
DEC '21: 1.4240437158469945


En 2024, la razón baja a 0.66, indicando que tiene menos efectivo disponible para pagar deudas inmediatas. En 2023, 2022 y 2021, la razón era más alta, lo que sugiere que la empresa tenía una mayor capacidad para cubrir sus obligaciones con efectivo. La caída en 2024 puede reflejar una disminución en la liquidez.

### Razón de Capital de Trabajo

In [106]:
razon_cap_trabajo = total_assets - total_liabilities

# resultados
print("Razón Capital de Trabajo para cada año:")
for year, ratio in zip(razon_cap_trabajo.index, razon_cap_trabajo):
    print(f"{year}: {ratio}")

Razón Capital de Trabajo para cada año:
DEC '24: -142.60000000000002
DEC '23: 95.89999999999998
DEC '22: 440.9
DEC '21: 343.70000000000005


En 2024, es negativa (-142.6), lo que significa que la empresa tiene más deudas a corto plazo que activos disponibles para cubrirlas, lo que puede ser preocupante.

## Razones de Rentabilidad